providers/gcs/gcs.go

```diff
@@ -91,31 +91,35 @@ func (b *Bucket) Name() string {
 	return b.name
 }
 
-// Iter calls f for each entry in the given directory. The argument to f is the full
-// object name including the prefix of the inspected directory.
-func (b *Bucket) Iter(ctx context.Context, dir string, f func(name string, attrs objstore.ObjectAttributes) error, options ...objstore.IterOption) error {
+func (b *Bucket) SupportedIterOptions() []objstore.IterOptionType {
+	return []objstore.IterOptionType{objstore.Recursive, objstore.UpdatedAt}
+}
+
+func (b *Bucket) IterWithAttributes(ctx context.Context, dir string, f func(attrs objstore.IterObjectAttributes) error, options ...objstore.IterOption) error {
+	for _, opt := range options {
+		if !slices.Contains(b.SupportedIterOptions(), opt.Type) {
```

Comment:
Since `SupportedIterOptions` is part of the `Bucket` interface, could we have the check in one place only to avoid copying the check everywhere? Perhaps we can have an internal type which wraps each bucket and performs the check?

Reply:
moved it to a common function to reduce the duplicate code.
I tried moving this to a wrapper or embed it within each bucket. But both the wrapper and embedding has to be explicitly added in every place a bucket instance is created and i think it is easy to miss doing this when someone changes the code and adds a new pathway that creates a bucket instance